lib/mandate/call_injector.rb

```diff
@@ -7,7 +7,12 @@ def self.extended(base)
       #   Foobar.new(some, args).call()
       class << base
         def call(*args)
-          new(*args).call
+          if args.last.is_a?(Hash) && 
+            :key == instance_method(:initialize).parameters.last&.first
+            new(*args[0..-2], **args[-1]).call
+          else
+            new(*args).call
+          end
```

Comment:
Hm, it looks kind of complicated. Why would the first part of the last part of the parameters of the instance method `:initialize` be the value `:key`? Maybe a short explanation in comments would help, or you could pull out named variables that signify what is what.

Alternatively, why not add `**kwargs` to the `call` method as well?

``` ruby
def initialize(*args, **kwargs)
  puts [args, kwargs].inspect
end

def call(*args, **kwargs)
  initialize(*args, **kwargs)
end

call("one", "two")
call("one", two: "three")
# >> [["one", "two"], {}]
# >> [["one"], {:two=>"three"}]
```

Reply:
> Alternatively, why not add **kwargs to the call method as well?

Because, annoyingly, if the last argument is a hash it will then get exploded into keywords. Or it will blow up. One of the two - but that's what the warning is about. It's annoying.

Reply:
> Why would the first part of the last part of the parameters of the instance method :initialize be the value :key?

That signifies that the last parameter is a set of keyword arguments. I'll add a comment.

Reply:
Also, very happy for a better implementation if you can find one that passes the tests with no warnings. I think this code is pretty ugly :)

Reply:
Hm, bummer. I don't really have a good solution for you, I admit. A lot of people seem to be complaining, so maybe the ruby core team will come up with some workaround one of these days, but ¯\_(ツ)_/¯. I guess as long as you can get it to work, it's probably as good as it'll get.